extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsConfig.java

```diff
@@ -95,6 +95,20 @@
 
     public RestClientLoggingConfig logging;
 
+    /**
+     * Global default connect timeout for automatically generated REST Clients. The attribute specifies a timeout
+     * in milliseconds that a client should wait to connect to the remote endpoint.
+     */
+    @ConfigItem(defaultValue = "15000", defaultValueDocumentation = "15000 ms")
+    public Optional<Long> connectTimeout = Optional.empty();
```

Comment:
Do we want an `Optional` here if we have a default value? The Optional will always have a value unless the user sets the key as empty.